api/base.proto

```diff
@@ -149,3 +150,16 @@ message ConfigSource {
     AggregatedConfigSource ads = 3;
   }
 }
+
+// Configuration for low level socket in listeners and clusters. If the configuration is empty,
+// the default low level socket configuration will be choosed based on the platform and
+// existence of tls_context.
+message LowLevelSocket {
+  // The name of low level socket to instantiate. The name must match a supported low level socket
+  // implementation.
+  string name = 1;
+
+  // Implementation specific configuration which depends on the implementation being instantiated.
+  // See the supported low level socket implementations for further documentation.
+  google.protobuf.Any config = 2;
```

Comment:
This is not very useful from a consumer perspective, especially since I have no idea what goes into this field. Sorry to be a PITA, but could you add some example (like the way we we indent yaml snippet in istio) or something? How do we populate this, if we were to handle this from LDS management plane?

Reply:
This is also a place where we are going to introduce inconsistency in the API. I do agree with @lizan that `Any` is the cleanest way to embed config, since it can support efficient binary representation and also a JSON-like `Struct` embedding. That is, providing you support at a code level embedded `Struct` decoding to the expected proto (please do this!).

OTOH, every other API for plugins and opaque config in Envoy today is `Struct` based. So, the question to be addressed here is do we care about (1) consistency across the Envoy xDS APIs or (2) the cleanest way to represent opaque configs for (i) efficient binary encoding and (ii) nice text proto representation. @lizan is this a fair summary of the question being asked?

I would personally vote for consistency, but I will defer to what the rest of the community thinks is best.

Reply:
@rshriram the configuration is to allow plug-in additional low-level socket implementations, with additional config, a default ones will be chosen automatically so you don't to handle this from LDS/CDS, unless you have another low-level socket impl, and want to use that.

Just think this as a filter, if you didn't link any, you don't need to configure one.

Reply:
@htuch yes that is a fair summary.

cc/@mattklein123 what is the conclusion for envoyproxy/envoy#1680? If we are moving to `Any`, (regardless with interim `oneof` or not, or the priority of it), then I'd prefer `Any` here. If everything is going to stick with `Struct` then I'm fine with `oneof` or `Struct`. I'd defer to you guys to make the call.

Reply:
All things being equal I would prefer consistency. In this case, perf does not matter, so I think struct is fine. So I would vote struct.